src/Frontend/actions/LocaleActions.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of the Sententiaregum project.
+ *
+ * (c) Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ * (c) Ben Bieler <benjaminbieler2014@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+import AppDispatcher from '../dispatcher/AppDispatcher';
+import LocaleConstants from '../constants/Locale';
+import {Locale, ApiKey} from '../util/http/facade/HttpServices';
+import reqwest from 'reqwest';
+
+/**
+ * Action creator which dispatches actions for the locale switcher.
+ *
+ * @author Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ */
+class LocaleActions {
+  /**
+   * Loads supported items and dispatch them to the store.
+   */
+  loadLanguages() {
+    reqwest({
+      url:     '/api/locale.json',
+      method:  'GET',
+      success: (response) => {
+        AppDispatcher.dispatch({
+          event:  LocaleConstants.GET_LOCALES,
+          result: response
+        });
+      }
+    });
+  }
+
+  /**
+   * Changes the locale.
+   *
+   * @param {string} locale
+   */
+  changeLocale(locale) {
+    Locale.setLocale(locale);
+
+    if (ApiKey.isLoggedIn()) {
+      reqwest({
```

Comment:
this should be executed before the locale will be changed.
As the reqwest task will be executed asynchronously this won't have any impact, but it should be called before the locale change